records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,213 @@
+from records_mover.utils import quiet_remove
+from records_mover.records.hints import cant_handle_hint
+from typing import TypedDict, Optional, Set, Dict, Any, Literal, NamedTuple
+from records_mover.records import DelimitedRecordsFormat
+from records_mover.records.types import (
+    HintEncoding, HintRecordTerminator,
+    HintFieldDelimiter, HintQuoteChar,
+    HintQuoting, HintEscape,
+    HintHeaderRow, HintCompression,
+    HintDoublequote,
+)
+
+# http://dev.mysql.com/doc/refman/8.0/en/charset-mysql.html
+MySqlCharacterSet = Literal['big5', 'binary', 'latin1', 'ucs2',
+                            'utf8', 'utf8mb4', 'utf16', 'utf16le',
+                            'utf32']
+
+MYSQL_CHARACTER_SETS_FOR_LOAD: Dict[HintEncoding, MySqlCharacterSet] = {
+    "UTF8": 'utf8',
+    "UTF16": 'utf16',
+    "UTF16LE": 'utf16le',
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-unicode.html
+    #
+    # The MySQL implementation of UCS-2, UTF-16, and UTF-32 stores
+    # characters in big-endian byte order and does not use a byte
+    # order mark (BOM) at the beginning of values. Other database
+    # systems might use little-endian byte order or a BOM. In such
+    # cases, conversion of values will need to be performed when
+    # transferring data between those systems and MySQL. The
+    # implementation of UTF-16LE is little-endian.
+    "UTF16BE": 'utf16',
+    # MySQL uses no BOM for UTF-8 values.
+    "LATIN1": 'latin1',
+    #  MySQL's latin1 is the same as the Windows cp1252 character set.
+    #
+    # https://dev.mysql.com/doc/refman/8.0/en/charset-we-sets.html
+    "CP1252": 'latin1',
+}
+
+
+# Mark as total=False so we can create this incrementally
+class MySqlLoadOptions(NamedTuple):
+    character_set: MySqlCharacterSet
+    field_terminator: str  # default '\t'
+    field_enclosed_by: Optional[str]  # default ''  # TODO: should this be optional?
+    field_optionally_enclosed_by: Optional[str]  # default None
+    field_escaped_by: Optional[str]  # default '\'
+    line_starting_by: str  # default ''
+    line_terminated_by: str  # default '\n'
+    ignore_n_lines: int
+
+    def generate_load_data_sql(self, filename: str) -> str:
+        # TODO: Base example where almost everything set
+        # TODO: Field enclosed by set to None
+        # TODO: Field optionally enclosed by set to None
+        # TODO: Field enclosed by and Field optionally enclosed by set to None
+        # TODO: Line starting by set to ''
+        # TODO: Line starting by set to something real ''
+        # TODO: Line terminated by set to '\n'
+        # TODO: Line terminated by set to other control characters (look up docs)
+        # TODO: Line terminated by set to something else
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker